From dc0c047ad6b9ac67ea97209048ea98adfdaf94a7 Mon Sep 17 00:00:00 2001
From: Owen D'Aprile <git@owen.sh>
Date: Tue, 27 Sep 2022 21:23:12 -0400
Subject: [PATCH] linux: add AppStream metadata

Adds an AppStream metainfo file that is used to display the app in
graphical app stores on Linux. This is also required for submission to
Flathub.
---
 .../org.thentrythis.Samplebrain.metainfo.xml  | 42 +++++++++++++++++++
 samplebrain.pro                               |  4 +-
 2 files changed, 45 insertions(+), 1 deletion(-)
 create mode 100644 desktop/org.thentrythis.Samplebrain.metainfo.xml

diff --git a/desktop/org.thentrythis.Samplebrain.metainfo.xml b/desktop/org.thentrythis.Samplebrain.metainfo.xml
new file mode 100644
index 0000000..63babad
--- /dev/null
+++ b/desktop/org.thentrythis.Samplebrain.metainfo.xml
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.thentrythis.Samplebrain</id>
+
+  <name>Samplebrain</name>
+  <summary>A custom sample mashing app designed by Aphex Twin</summary>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0-only</project_license>
+
+  <description>
+    <p>
+      Samplebrain chops samples up into a &apos;brain&apos; of interconnected small sections called blocks which are connected into a network by similarity. It processes a target sample, chopping it up into blocks in the same way, and tries to match each block with one in it&apos;s brain to play in realtime.
+    </p>
+    <p>
+      This allows you to interpret a sound with a different one. As we worked on it (during 2015 and 2016) we gradually added more and more tweakable parameters until it became slightly out of control.
+    </p>
+  </description>
+
+  <launchable type="desktop-id">samplebrain.desktop</launchable>
+
+  <categories>
+    <category>AudioVideo</category>
+    <category>Music</category>
+  </categories>
+
+  <content_rating type="oars-1.0"></content_rating>
+
+  <url type="homepage">https://thentrythis.org/projects/samplebrain/</url>
+  <url type="bugtracker">https://gitlab.com/then-try-this/samplebrain/-/issues</url>
+  <url type="vcs-browser">https://gitlab.com/then-try-this/samplebrain</url>
+
+  <screenshots>
+    <screenshot type="default">
+      <image type="source" width="1280" height="720">https://gitlab.com/then-try-this/samplebrain/-/raw/main/desktop/screenshots/01-main-with-project.png</image>
+    </screenshot>
+  </screenshots>
+
+  <releases>
+    <release version="0.18" date="2022-09-22"/>
+  </releases>
+</component>
diff --git a/samplebrain.pro b/samplebrain.pro
index c7df563..22799fb 100644
--- a/samplebrain.pro
+++ b/samplebrain.pro
@@ -61,6 +61,8 @@ unix:desktopfile.path = $$PREFIX/share/applications/
 unix:desktopfile.files = desktop/samplebrain.desktop
 unix:iconfile.path = $$PREFIX/share/icons/hicolor/scalable/apps
 unix:iconfile.files = desktop/samplebrain.svg
+unix:metainfofile.path = $$PREFIX/share/metainfo
+unix:metainfofile.files = desktop/org.thentrythis.Samplebrain.metainfo.xml
 
 target.path = $$PREFIX/bin
-INSTALLS += target desktopfile iconfile
+INSTALLS += target desktopfile iconfile metainfofile
-- 
GitLab

